admin_bans.php

```diff
@@ -280,9 +280,13 @@
 		$ban_expire = 'NULL';
 
 	$ban_user = ($ban_user != '') ? '\''.$db->escape($ban_user).'\'' : 'NULL';
-	$ban_ip = ($ban_ip != '') ? '\''.$db->escape($ban_ip).'\'' : 'NULL';
+	$ban_ip = ($ban_ip != '' && $ban_ip != '127.0.0.1' && $ban_ip != '::1') ? '\''.$db->escape($ban_ip).'\'' : 'NULL';
 	$ban_email = ($ban_email != '') ? '\''.$db->escape($ban_email).'\'' : 'NULL';
 	$ban_message = ($ban_message != '') ? '\''.$db->escape($ban_message).'\'' : 'NULL';
+	
+	// Don't insert an empty record
+	if ($ban_user == 'NULL' && $ban_ip == 'NULL' && $ban_email == 'NULL' && $ban_message == 'NULL')
+		redirect('admin_bans.php', $lang_admin_bans['Ban added redirect']);

```

Comment:
It's not, but since this was not taken care of for years, I thought we could avoid to add a string in language files and keep the same behaviour as before. If you disagree, I'll add a language string.